### PR TITLE
Remove extra column in econ view

### LINKED ISSUE #1818: The Economics View has an extra unused header column
**Describe the bug**
The Economics View has an extra header column that doesn't show any data and if clicked will expand to fill in the width of the table.

**To Reproduce**
1. Open a game
2. Click on economics view
3. Look at the right edge of the table header and notice the extra "box", click on it

**Expected behavior**
No extraneous columns in a table

**Screenshots**
N/A

**Platform and version (please complete the following information):**
 - OS: Linux, but assume all
 - Freeciv21 version: 3.0-rc.2+
 - Ruleset/Longturn game (if applicable): n/a

**Additional context**
Will also show in new Units View (#1808) when merged as it was based on econ view.

### PR DESCRIPTION
**Pull request by jwrober**
Do some other cleanup items too such as making the icon bigger, centering it in the cell and set alternating row colors. 

Closes #1818

### CODE REVIEW

File: client/views/view_economics.cpp [hunk 1]
```diff
@@ -31,9 +31,10 @@ eco_report::eco_report() : QWidget()
 
   QStringList slist;
   slist << _("Type") << Q_("?Building or Unit type:Name") << _("Redundant")
-        << _("Count") << _("Cost") << _("U Total") << QLatin1String("");
+        << _("Count") << _("Cost") << _("Upkeep Total");
```
Comment (lmoureaux) [inline client/views/view_economics.cpp:34]: Wondering what sounds best, `Total Upkeep` or `Upkeep Total`?

File: client/views/view_economics.cpp [hunk 2]
```diff
@@ -90,32 +91,37 @@ void eco_report::update_report()
     ui.eco_widget->insertRow(i);
     for (j = 0; j < 6; j++) {
       item = new QTableWidgetItem;
+      QLabel *lbl = new QLabel;
```
Comment (lmoureaux) [inline client/views/view_economics.cpp:94]: You leak the label when `j != 0` (memory leak): it is neither deleted nor used. Simply create it when getting the sprite.

File: client/views/view_economics.cpp [hunk 3]
```diff
@@ -124,40 +130,44 @@ void eco_report::update_report()
...
+    auto sprite = get_unittype_sprite(tileset, putype, direction8_invalid())
+                      ->scaledToHeight(h);
+
+    cid id = cid_encode_unit(putype);
 
     ui.eco_widget->insertRow(i + max_row);
     for (j = 0; j < 6; j++) {
       item = new QTableWidgetItem;
-      item->setTextAlignment(Qt::AlignHCenter);
+      QLabel *lbl = new QLabel;
```
Comment (lmoureaux) [inline client/views/view_economics.cpp:142]: Another leaked label

File: client/views/view_economics.cpp (hunk 2 above)
Comment (jwrober) [reply client/views/view_economics.cpp:94]: How is that different than the `QTableWidgetItem`?

File: client/views/view_economics.cpp (hunk 1 above)
Comment (jwrober) [reply client/views/view_economics.cpp:34]: I think its probably better as "Total Upkeep" I'll change

**Pull request merged**